api/net/ip4/udp.hpp

```diff
@@ -141,27 +150,52 @@ namespace net {
     public:
       Port_in_use_exception(UDP::port_t p)
         : port_(p) {}
-      virtual const char* what() const noexcept {
-        return "UDP port already in use";
-      }
 
-      UDP::port_t port(){
-        return port_;
-      }
+      virtual const char* what() const noexcept
+      { return "UDP port already in use"; }
+
+      UDP::port_t port()
+      { return port_; }
 
     private:
       UDP::port_t port_;
     };
 
   private:
+    static constexpr uint16_t exp_t_ {60 * 5};
 
-    downstream  network_layer_out_;
-    Stack&      stack_;
+    std::chrono::minutes        flush_interval_{5};
+    downstream                  network_layer_out_;
+    Stack&                      stack_;
     std::map<port_t, UDPSocket> ports_;
-    port_t      current_port_;
+    port_t                      current_port_;
 
     // the async send queue
     std::deque<WriteBuffer> sendq;
+
+    /** Error entries are just error callbacks and timestamps */
+    class Error_entry {
+    public:
+      Error_entry(UDP::error_handler cb) noexcept
+      : callback(std::move(cb)), timestamp(RTC::time_since_boot())
+      {}
+
+      bool expired() noexcept
+      { return timestamp + exp_t_ < RTC::time_since_boot(); }
+
+      UDP::error_handler callback;
+
+    private:
+      RTC::timestamp_t timestamp;
+
+    }; //< class Error_entry
+
+    /** The error callbacks that the user has sent in via the UDPSockets' sendto and bcast methods */
+    std::unordered_map<Socket, Error_entry, Socket::pair_hash> error_callbacks_;
```

Comment:
A implementation of `std::hash<Socket>` would make it possible to drop the third template arg `Socket::pair_hash` (this function can be called inside the std::hash`). See http://en.cppreference.com/w/cpp/utility/hash